source/grid/manifold_lib.cc

```diff
@@ -373,6 +373,52 @@ get_tangent_vector (const Point<spacedim> &p1,
 
 
 
+template<int dim, int spacedim>
+Tensor<1, spacedim>
+SphericalManifold<dim, spacedim>::
+normal_vector (const typename Triangulation<dim, spacedim >::face_iterator &/*face*/,
+               const Point<spacedim>                                       &p) const
+{
+  const Tensor<1,spacedim> unnormalized_normal = p-center;
+  return unnormalized_normal/unnormalized_normal.norm();
```

Comment:
Would it make sense to add an assertion that the face is indeed on the surface, i.e., by checking that the distance of all vertices on the face is within say 10% of the distance of the first point?

Reply:
10% is a very large tolerance. How about tying it somehow to the distance of the vertices?

Reply:
That's right, 10% of the distance between vertex 0 and 1 (or the minimum vertex distance) would be much more appropriate.

Reply:
Oh, I thought you had meant "distance from the origin", as opposed to "distance from each other".